json-rpc/src/tests/utils.rs

```diff
@@ -383,6 +387,52 @@ impl DbReader for MockDiemDB {
     }
 }
 
+impl MoveStorage for MockDiemDB {
```

Comment:
seems sad to re-implement logic here.
Maybe a decorator is better?
Like:  MoveDbReader: MoveStorage { dbreader: DbReader }

Reply:
Agreed re: the sadness, I'll see if I can make the decorator work.

Reply:
I tried, but couldn't do this without adding new public types in the storage interfaces that were only used in this test code. Stuck with the code duplication instead.